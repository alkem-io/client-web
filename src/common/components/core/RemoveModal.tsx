import React, { FC } from 'react';
import { useTranslation } from 'react-i18next';
import Dialog from '@mui/material/Dialog';
import WrapperButton from './WrapperButton';
import { DialogActions, DialogContent, DialogTitle } from './dialog';

interface RelationRemoveModalProps {
  show: boolean;
  onCancel: () => void;
  text: string;
  onConfirm: () => void;
  title?: string;
}

const RemoveModal: FC<RelationRemoveModalProps> = ({ show, text, onCancel, title, onConfirm }) => {
  const { t } = useTranslation();
  const defaultTitle = t('components.remove-modal.title');

  return (
    <Dialog open={show} maxWidth="md" fullWidth aria-labelledby="remove-dialog-title">
      <DialogTitle id="remove-dialog-title" onClose={onCancel}>
        {title || defaultTitle}
      </DialogTitle>
      <DialogContent dividers>{text}</DialogContent>
      <DialogActions>
        <WrapperButton small variant={'negative'} onClick={onConfirm} text={t('buttons.remove')} />
      </DialogActions>
    </Dialog>
  );
};

export default RemoveModal;
