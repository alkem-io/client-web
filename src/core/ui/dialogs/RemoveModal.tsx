import { DialogContent } from '@mui/material';
import Dialog from '@mui/material/Dialog';
import { useTranslation } from 'react-i18next';
import NegativeButton from '../button/NegativeButton';
import { DialogActions, DialogTitle } from '../dialog/deprecated';

type RelationRemoveModalProps = {
  show: boolean;
  onCancel: () => void;
  text: string;
  onConfirm: () => void;
  title?: string;
};

const RemoveModal = ({ show, text, onCancel, title, onConfirm }: RelationRemoveModalProps) => {
  const { t } = useTranslation();
  const defaultTitle = t('components.remove-modal.title');

  return (
    <Dialog open={show} maxWidth="md" fullWidth aria-labelledby="remove-dialog-title">
      <DialogTitle id="remove-dialog-title" onClose={onCancel}>
        {title || defaultTitle}
      </DialogTitle>
      <DialogContent dividers>{text}</DialogContent>
      <DialogActions>
        <NegativeButton onClick={onConfirm}>{t('buttons.remove')}</NegativeButton>
      </DialogActions>
    </Dialog>
  );
};

export default RemoveModal;
