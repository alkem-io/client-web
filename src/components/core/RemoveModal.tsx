import React, { FC } from 'react';
import { useTranslation } from 'react-i18next';
import Dialog from '@material-ui/core/Dialog';
import Button from './Button';
import { DialogActions, DialogContent, DialogTitle } from '../core/dialog';

interface RelationRemoveModalProps {
  show: boolean;
  onCancel: () => void;
  text: string;
  onConfirm: () => void;
  title?: string;
}

const RemoveModal: FC<RelationRemoveModalProps> = ({ show, text, onCancel, title = 'Confirm remove', onConfirm }) => {
  const { t } = useTranslation();
  return (
    <Dialog open={show} maxWidth="md" fullWidth aria-labelledby="remove-dialog-title">
      <DialogTitle id="remove-dialog-title" onClose={onCancel}>
        {title}
      </DialogTitle>
      <DialogContent dividers>{text}</DialogContent>
      <DialogActions>
        <Button small variant={'negative'} onClick={onConfirm} text={t('buttons.remove')} />
      </DialogActions>
    </Dialog>
  );
};

export default RemoveModal;
