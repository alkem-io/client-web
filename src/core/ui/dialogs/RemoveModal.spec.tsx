/**
 * @jest-environment jsdom
 */
import userEvent from '@testing-library/user-event';
import { render, screen } from '../../../main/test/testUtils';
import RemoveModal from './RemoveModal';
import { expect, test, describe } from 'vitest';

describe('RemoveModal - main', () => {
  const show = true;
  const text = 'text';
  const title = 'confirm';
  const onCancel = () => {};
  const onConfirm = () => {};

  test('renders correct RemoveModal component', () => {
    // arrange
    const { asFragment } = render(<RemoveModal show={show} onCancel={onCancel} onConfirm={onConfirm} text={text} />);

    // act
    const html = asFragment();

    // assert
    expect(html).toMatchSnapshot();
  });

  test.skip('renders correct modal text', () => {
    // arrange
    render(<RemoveModal show={show} onCancel={onCancel} onConfirm={onConfirm} text={text} />);

    // assert
    expect(screen.getByText(text)).toBeInTheDocument();
  });

  test.skip('renders correct modal title', () => {
    // arrange
    render(<RemoveModal show={show} onCancel={onCancel} onConfirm={onConfirm} text={text} title={title} />);

    // assert
    expect(screen.getByText(title)).toBeInTheDocument();
  });

  test.skip('calls modal cancel button', () => {
    // arrange
    render(<RemoveModal show={show} onCancel={onCancel} onConfirm={onConfirm} text={text} />);

    // act
    userEvent.click(screen.getByText('Remove'));

    // assert
    expect(onCancel).toHaveBeenCalled(0);
    expect(onConfirm).toHaveBeenCalled(1);
  });

  test.skip('calls modal remove button', () => {
    // arrange
    render(<RemoveModal show={show} onCancel={onCancel} onConfirm={onConfirm} text={text} />);

    // act
    userEvent.click(screen.getByText('Remove'));

    // assert
    expect(onCancel).toHaveBeenCalled(0);
    expect(onConfirm).toHaveBeenCalled(1);
  });
});

describe('RemoveModal - negative', () => {
  test.skip('modal is not loaded', () => {
    // arrange
    const show = false;
    const text = 'experiment';
    const title = 'confirm';
    const onCancel = () => {};
    const onConfirm = () => {};
    render(<RemoveModal show={show} onCancel={onCancel} onConfirm={onConfirm} text={text} title={title} />);

    // assert
    expect(screen.queryAllByRole('dialog')).toHaveLength(0);
  });
});
