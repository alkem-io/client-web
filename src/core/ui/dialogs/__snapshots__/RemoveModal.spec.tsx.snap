// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RemoveModal - main > renders correct RemoveModal component 1`] = `<DocumentFragment />`;
